### PR TITLE
Add narrative documentation for physical types

### LINKED ISSUE #11588: Add narrative documentation on physical types
### Description

In #11204 we made the `physical_type` attribute of units into instances of the new [`PhysicalType`](https://github.com/astropy/astropy/blob/4a289db09ac1026b038d156c1e1139223a762590/astropy/units/physical.py#L199) class.  Before this, `physical_type` returned a string.  It's now possible to do dimensional analysis along the lines of:
```Python
>>> import astropy.units as u
>>> length = u.m.physical_type
>>> area = u.get_physical_type("area")
>>> length * area
PhysicalType('volume')
```
There are descriptions on how to use physical types in the docstrings for `PhysicalType`, `get_physical_type`, and `UnitBase.physical_type`.  Following a [suggestion](https://github.com/astropy/astropy/pull/11204#pullrequestreview-636754774) in #11204, it would be helpful to add a discussion of physical types in the narrative documentation for `astropy.units`.  

It might be helpful to write out or have a table of all of the names of physical types.  I'm not sure if this would be best in the narrative documentation or in the docstring for `get_physical_type`.

### PR DESCRIPTION
**Pull request by namurphy**
### Description

This pull request is to add narrative documentation on how to use the `PhysicalType` instances that were added in #11204.

Fixes #11588.

### CODE REVIEW

**Comment by pllim** [comment]
Not necessarily this particular example per se, but maybe include an example where a unit can have multiple physical types? I ran into this one in my downstream package:

```python
>>> from astropy import units as u
>>> FLAM = u.def_unit(
...     'flam', u.erg / (u.cm**2 * u.s * u.AA),
...     format={'generic': 'FLAM', 'console': 'FLAM'})
>>> FLAM.physical_type
PhysicalType({'power density', 'spectral flux density wav'})
```

**Review by mhvk — COMMENTED** [review]
Thanks, very nice! 

I think it is good to keep it simple (as you did), but it makes sense to, as @pllim suggested, also have one example of a unit whose physical type is ambiguous (maybe just pressure?)

**Comment by namurphy** [comment]
> Not necessarily this particular example per se, but maybe include an example where a unit can have multiple physical types? 

Agreed!  I still have to write that part, as well as an example on dimensional analysis with `PhysicalType` instances.

File: docs/units/physical_types.rst [hunk 1]
```diff
@@ -0,0 +1,56 @@
+.. |PhysicalType| replace:: :class:`~astropy.units.PhysicalType`
+.. |Quantity| replace:: :class:`~astropy.units.Quantity`
```
Comment (nstarman) [inline docs/units/physical_types.rst:2]: With #11599 in imminently, |Quantity| is already taken care of. |PhysicalType| might be a good addition to conf.py as well

Comment (mhvk) [reply docs/units/physical_types.rst:2]: #11599 is now merged - agreed that `PhysicalType` might as well be in `conf.py` as well.

**Review by mhvk — APPROVED** [review]
This looks good!

File: docs/units/physical_types.rst [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+========================
+
+.. EXAMPLE START: Accessing Physical Types
+
+Using `~astropy.units.get_physical_type` lets us acquire |PhysicalType|
+instances from strings with a name of a physical type, units, |Quantity|
+instances, objects that can become quantities (e.g., numbers), and
+|PhysicalType| instances.
+
+  >>> u.get_physical_type('speed')  # from the name of a physical type
```
Comment (mhvk) [inline docs/units/physical_types.rst:19]: Need to `import astropy.units as u` on top here to get rid of the errors

File: docs/units/physical_types.rst [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+|PhysicalType| instances support multiplication, division,
+and exponentiation. Because of this, they can be used for
+dimensional analysis.
+
+  >>> import astropy.units as u
+  >>> length = u.get_physical_type('length')
+  >>> time = u.get_physical_type('time')
+  >>> length ** 2
+  PhysicalType({'area')
```
Comment (mhvk) [inline docs/units/physical_types.rst:89]: The curly bracket here causes the tests to fail...

**Comment by nstarman** [comment]
As I did in perhaps a bad way, and @mhvk suggested an alternative in #11595, it would be great to also have a list of the valid physical types in the docs. I agree that @mhvk's proposal is more in line with current astropy conventions... No problems if you don't want to do it here, I'll tackle it at some point in #11595, but then it won't be in v4.3!

**Comment by nstarman** [comment]
> As I did in perhaps a bad way, and @mhvk suggested an alternative in #11595, it would be great to also have a list of the valid physical types in the docs. I agree that @mhvk's proposal is more in line with current astropy conventions... No problems if you don't want to do it here, I'll tackle it at some point in #11595, but then it won't be in v4.3!

never mind. @mhvk is tackling this already!

**Review by nstarman — APPROVED** [review]

**Review by mhvk — APPROVED** [review]
Great!  Let's just let CI run just to be sure, but otherwise ready to go in. Thanks, @namurphy!

**Pull request merged**